### PR TITLE
SONAR-7968 Create /api/settings/list_definitions WS

### PR DESCRIPTION
**Pull request by julienlancelot**

### CODE REVIEW

File: server/sonar-server/src/main/java/org/sonar/server/settings/ws/ListDefinitionsAction.java [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+    List<PropertyFieldDefinition> fields = definition.fields();
+    if (!fields.isEmpty()) {
+      fields.stream()
+        .filter(fieldDefinition -> !fieldDefinition.type().equals(PropertyType.LICENSE))
+        .forEach(fieldDefinition -> addField(fieldDefinition, builder));
+    }
+  }
+
+  private void addField(PropertyFieldDefinition fieldDefinition, Settings.Definition.Builder builder) {
+    builder.addFieldsBuilder()
```
Comment (SonarTech) [inline server/sonar-server/src/main/java/org/sonar/server/settings/ws/ListDefinitionsAction.java]: MAJOR Make "addField" a "static" method. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS2325)

**Comment by SonarTech** [comment]
SonarQube analysis reported 1 issue
* MAJOR 1 major

Watch the comments in this conversation to review them.

File: server/sonar-server/src/main/java/org/sonar/server/settings/ws/ListDefinitionsAction.java [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+        "<ul>" +
+        "<li>'Administer System'</li>" +
+        "<li>'Administer' rights on the specified project</li>" +
+        "</ul>", PARAM_COMPONENT_ID, PARAM_COMPONENT_KEY))
+      .setResponseExample(getClass().getResource("list_definitions-example.json"))
+      .setSince("6.1")
+      .setHandler(this);
+
+    action.createParam(PARAM_COMPONENT_ID)
+      .setDescription("Project or module id")
```
Comment (stas-vilchik) [inline server/sonar-server/src/main/java/org/sonar/server/settings/ws/ListDefinitionsAction.java]: > component id

File: server/sonar-server/src/main/java/org/sonar/server/settings/ws/ListDefinitionsAction.java [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+      .setExampleValue(UUID_EXAMPLE_01);
+
+    action.createParam(PARAM_COMPONENT_KEY)
+      .setDescription("Project or module key")
```
Comment (stas-vilchik) [inline server/sonar-server/src/main/java/org/sonar/server/settings/ws/ListDefinitionsAction.java]: > component id

File: server/sonar-server/src/test/java/org/sonar/server/settings/ws/ListDefinitionsActionTest.java [hunk 4]
```diff
@@ -0,0 +1,356 @@
...
+import static org.sonar.core.permission.GlobalPermissions.DASHBOARD_SHARING;
+import static org.sonar.core.permission.GlobalPermissions.SYSTEM_ADMIN;
+import static org.sonar.db.component.ComponentTesting.newProjectDto;
+import static org.sonarqube.ws.Settings.Type.BOOLEAN;
+import static org.sonarqube.ws.Settings.Type.PROPERTY_SET;
+import static org.sonarqube.ws.Settings.Type.SINGLE_SELECT_LIST;
+import static org.sonarqube.ws.Settings.Type.STRING;
+import static org.sonarqube.ws.Settings.Type.TEXT;
+
+public class ListDefinitionsActionTest {
```
Comment (stas-vilchik) [inline server/sonar-server/src/test/java/org/sonar/server/settings/ws/ListDefinitionsActionTest.java]: add tests:

- test response example
- fail when anonymous

**Comment by julienlancelot** [comment]
Thanks @stas-vilchik ! I've added test on response example and on permission check.

**Pull request merged**